Compass/LoginViewController.swift

```diff
@@ -0,0 +1,216 @@
+//
+//  LoginViewController.swift
+//  Compass
+//
+//  Created by Csabi on 08/10/2018.
+//  Copyright © 2018 Csabi. All rights reserved.
+//
+
+import UIKit
+import Parse
+
+class LoginViewController: UIViewController {
+    
+    var isloginPage = true
+    var activityIndicator = UIActivityIndicatorView()
+    
+    let titleLabel = UILabel(frame: CGRect.zero)
+    let NametextField = UITextField(frame: CGRect.zero)
+    let passTextField = UITextField(frame: CGRect.zero)
+    let loginButton = UIButton(frame: CGRect.zero)
+    let switchPageButton = UIButton(frame: CGRect.zero)
+    let par = PServer()
+    let userData = UserData()
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        par.initParse(appID: "492795c6ea25112881915677092fb19d95f43ce0", clKey: "6c4448eb0dc5d344a0ca35f8d8f978ff82b76028", serverAddress: "http://18.188.82.67:80/parse")
+        
+        navigationController?.navigationBar.barTintColor = UIColor.black
+        self.navigationController?.navigationBar.titleTextAttributes = [NSAttributedStringKey.foregroundColor: UIColor.green]
+        navigationItem.title = "No storyboard"
+        self.view.backgroundColor = UIColor.darkGray
+        
+        titleLabel.textColor = UIColor.red
+        
+        loginButton.addTarget(self, action: #selector(loginButtonPressed), for: .touchUpInside)
+        
+        switchPageButton.setTitle("Not registered? SignUp", for: .normal)
+        switchPageButton.addTarget(self, action: #selector(SwitchPagePressed), for: .touchUpInside)
+        
+        NametextField.placeholder = "e-mail address..."
+        NametextField.text = "test@mail.co.uk"
+        NametextField.textColor = UIColor.white
+        NametextField.layer.backgroundColor = UIColor.blue.cgColor
+        NametextField.attributedPlaceholder = NSAttributedString(string: "Name...", attributes: [NSAttributedStringKey.foregroundColor: UIColor.white])
+        
+        
+        passTextField.placeholder = "password..."
+        passTextField.text = "pass"
+        passTextField.textColor = UIColor.white
+        passTextField.isSecureTextEntry = true
+        passTextField.layer.backgroundColor = UIColor.blue.cgColor
+        passTextField.attributedPlaceholder = NSAttributedString(string: "Password...", attributes: [NSAttributedStringKey.foregroundColor : UIColor.white])
+        
+        
+        let stackView = UIStackView(arrangedSubviews: [titleLabel, NametextField, passTextField, loginButton, switchPageButton])
+        stackView.axis = .vertical
+        stackView.spacing = 10;
+        
+        if isloginPage {
+            titleLabel.text = "Login Page"
+            loginButton.setTitle("Login", for: .normal)
+            switchPageButton.setTitle("Go to Signup page", for: .normal)
+            
+        } else {
+            titleLabel.text = "Signup Page"
+            loginButton.setTitle("Signup", for: .normal)
```

Comment:
I might be tempted to make separate buttons for login and signup, and to show and hide the appropriately. Since you using a stack view, it will handle the spacing automatically when you show or hide something.
The main advantage of this would be having separate functions for loginButtonPressed/signupButtonPressed which would avoid the big if/else you have in there